README.md

```diff
@@ -230,36 +265,92 @@ module.exports = {
 };
 ```
 
-Migration files should be located in the same directory, according to the info you gave to the `Umzug` constructor.
+Migration files can be located anywhere - they will typically be loaded according to a glob pattern provided to the `getUmzug` constructor constructor.
 
 #### Direct migrations list
 
-You can also specify directly a list of migrations to the `Umzug` constructor. We recommend the usage of the `Umzug.migrationsList()` function
-as bellow:
+You can also specify directly a list of migrations to the `getUmzug` constructor:
+
+```js
+const { getUmzug } = require('umzug');
+
+const umzug = getUmzug({
+  migrations: [
+    {
+      // the name of the migration is mandatory
+      name: '00-first-migration',
+      async up({ context }) { /* ... */ },
+      async down({ context }) { /* ... */ }
+    },
+    {
+      name: '01-foo-bar-migration',
+      async up({ context }) { /* ... */ },
+      async down({ context }) { /* ... */ }
+    }
+  ],
+  context: sequelize.getQueryInterface(),
+});
+```
+
+To load migrations in another format, you can use the `resolve` function:
 
 ```js
-const { Umzug, migrationsList } = require('umzug');
+const { getUmzug } = require('umzug');
+const fs = require('fs')
 
-const umzug = new Umzug({
-  migrations: migrationsList(
-    [
-      {
-        // the name of the migration is mandatory
-        name: '00-first-migration',
-        async up(queryInterface) { /* ... */ },
-        async down(queryInterface) { /* ... */ }
+const umzug = getUmzug({
+  migrations: {
+    glob: 'migrations/*.sql',
+    resolve: ({ name, path, context }) => ({
+      up: async () => {
+        const sql = fs.readFileSync(path)
+        return context.sequelize.query(sql)
       },
-      {
-        name: '01-foo-bar-migration',
-        async up(queryInterface) { /* ... */ },
-        async down(queryInterface) { /* ... */ }
+      down: async (({ name, path, context })) => {
+        const sql = fs.readFileSync(path.replace('.sql', '.down.sql'))
+        return context.sequelize.query(sql)
       }
-    ],
-    // an optional list of parameters that will be sent to the `up` and `down` functions
-    [
-      sequelize.getQueryInterface()
-    ]
-  )
+    })
+  },
+  context: sequelize.getQueryInterface(),
+});
+```
+
+### Upgrading from v2.x
```

Comment:
Fantastic! I really like the changes so far. Will take a closer look in the PR tomorrow.